addons/mail/static/tests/thread/unread_messages_banner.test.js

```diff
@@ -223,3 +225,49 @@ test("keep banner after mark as unread when scrolling to bottom", async () => {
     await scroll(".o-mail-Thread", "bottom");
     await contains(".o-mail-Thread-banner", { text: "30 new messages" });
 });
+
+test("sidebar and banner counters display same value", async () => {
+    const pyEnv = await startServer();
+    const bobPatnerId = pyEnv["res.partner"].create({ name: "Bob" });
+    const bobUserId = pyEnv["res.users"].create({ name: "Bob", partner_id: bobPatnerId });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "chat",
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ partner_id: bobPatnerId }),
+        ],
+    });
+    for (let i = 0; i < 30; ++i) {
+        pyEnv["mail.message"].create({
+            author_id: serverState.partnerId,
+            body: `message ${i}`,
+            model: "discuss.channel",
+            res_id: channelId,
+        });
+    }
+    await start();
+    await openDiscuss();
+    await contains(".o-mail-DiscussSidebar-badge", {
+        text: "30",
+        parent: [".o-mail-DiscussSidebarChannel", { text: "Bob" }],
+    });
+    await click(".o-mail-DiscussSidebarChannel", { text: "Bob" });
+    await contains(".o-mail-Thread-banner", { text: "30 new messages" });
+    await contains(".o-mail-Discuss-sidebarBadge", { text: "30", count: 0 });
```

Comment:
`.o-mail-Discuss-sidebarBadge` does not exist, I suppose you meant `.o-mail-DiscussSidebar-badge`.

This is the LOC I think the specs should be challenged: it's strange to have badge 30 then 0 (invisible) then 31. I'd say it should be:
- either 30 > 0 > 1
- or 30 > 31
The difference being whether the recent messages have been read or not

30 > 0 > 31 feels buggy,